### PR TITLE
Move gradle_non_android_plugin_test from its own test builders into tools integration.shard

### PR DESCRIPTION
**Pull request by jmagman**
`gradle_non_android_plugin_test` seems particularly wasteful of LUCI resources.  The test only takes 1:30-3:00 minutes to run, but each time it does it spins up an entire build that takes longer to set up and tear down than the test takes.

Move this test to the tools `integration.shard`.
https://ci.chromium.org/p/flutter/builders/try/Linux%20gradle_non_android_plugin_test/5263?
https://ci.chromium.org/p/flutter/builders/try/Mac%20gradle_non_android_plugin_test/5344?
https://ci.chromium.org/p/flutter/builders/try/Windows%20gradle_non_android_plugin_test/5261?

Also, this happened to make the unrelated `tool_backend_test` run first in its shard, and it didn't pass because the output contained `Building flutter tool`: 
```
Expected: 'PROJECT_DIR environment variable must be set to the location of Flutter project to be built.'
  Actual: 'Building flutter tool...\n'
            'PROJECT_DIR environment variable must be set to the location of Flutter project to be built.'
```
So fix that.

Part of https://github.com/flutter/flutter/issues/65790

### CODE REVIEW

File: dev/devicelab/bin/tasks/gradle_non_android_plugin_test.dart [hunk 1]
```diff
@@ -1,113 +0,0 @@
-// Copyright 2014 The Flutter Authors. All rights reserved.
```
Comment (christopherfujino) [inline dev/devicelab/bin/tasks/gradle_non_android_plugin_test.dart:1]: so to delete this file, you would have to remove the builder first. however, if you remove the builder, it won't run on older release branches, which won't have the tools version of the test.

instead, can you make this test file a no-op and instead mark the LUCI builder with a `# TODO(fujino): delete this once PR #80161 rolls to stable`?

**Review by blasten — APPROVED** [review]
LGTM % @christopherfujino comment

File: dev/devicelab/bin/tasks/gradle_non_android_plugin_test.dart (hunk 1 above)
Comment (jmagman) [reply dev/devicelab/bin/tasks/gradle_non_android_plugin_test.dart:1]: You mean add the comment in the `infra` star file?
https://github.com/flutter/infra/blob/707032677f8580d93ceafb4aa9a38aaa2b06615c/config/devicelab_config.star#L174-L175

**Review by christopherfujino — APPROVED** [review]
You can merge this PR as is. I'll add a TODO to the builder config.

**Comment by jmagman** [comment]
> You can merge this PR as is. I'll add a TODO to the builder config.

I'm happy to add it, do you mean add the comment to the `infra` star file?
https://github.com/flutter/infra/blob/707032677f8580d93ceafb4aa9a38aaa2b06615c/config/devicelab_config.star#L174-L175

**Pull request merged**